src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -288,9 +289,15 @@ def maybe_raise_or_warn(
         # Dynamo wraps the original model in a private class.
         # I didn't find a public API to get the original class.
         sub_model = passed_class_obj[name]
-        model_cls = sub_model.__class__
         if is_compiled_module(sub_model):
-            model_cls = sub_model._orig_mod.__class__
+            sub_model = sub_model._orig_mod
+
+        model_cls = sub_model.__class__
+        if USE_PEFT_BACKEND:
+            from peft import PeftModel
```

Comment:
Not sure if this condition is needed, I would rather do something like:
```suggestion
        if is_peft_available() and isinstance(sub_model, PeftModel):
            from peft import PeftModel
```
As we don't use `PeftModel` for the PEFT integration, it should be totally agnostic to the minimum PEFT version we require for the integration.